src/datatip.jl

```diff
@@ -0,0 +1,91 @@
+"""
+Regex to match code blocks from markdown texts.
+"""
+const codeblock_regex = r"```((?!```).)*?```"s
+"""
+Regex that matches for a Julia documentation text when there is no documentation
+is found for the target variable/method.
+Refered to https://github.com/JuliaLang/julia/blob/master/stdlib/REPL/src/docview.jl#L152.
+"""
+const nodoc_regex = r"No documentation found.\n\nBinding `.*` does not exist.\n"
+
+# Extract only code blocks from Markdown.MD
+function searchcodeblocks(docs)
+  codeblocks = []
+  searchcodeblocks(docs, codeblocks)
+  codeblocks
+end
+function searchcodeblocks(docs, codeblocks)
+  for content in docs.content
+    if content isa Markdown.Code
+      push!(codeblocks, content.code)
+    elseif content isa Markdown.MD
+      searchcodeblocks(content, codeblocks)
+    end
+  end
+end
+
+function processmdtext!(text, datatips)
+  (text == "" || text == "\n") && return
+  push!(datatips, Dict(:type  => :markdown,
+                       :value => text))
+end
+
+function processmdcode!(code, datatips)
+  push!(datatips, Dict(:type  => :snippet,
+                       :value => code))
+end
+
+function processmethodtable!(word, mtable, datatips)
+  isempty(mtable) && return
+
+  header = "\n***\n`$(word)` has **$(length(mtable))** methods\n"
+
+  body = map(mtable) do m
+    mstring = string(m)
+    text = mstring[1:match(r" at ", mstring).offset + 3]
+    isbase = m.module === Base || parentmodule(m.module) === Base
+    file = isbase ? basepath(string(m.file)) : m.file
+    # @NOTE: Datatip service component can't handle links
+    "- $(text)$(file):$(m.line)"
+  end |> lists -> join(lists, "\n")
+
+  push!(datatips, Dict(:type  => :markdown,
+                       :value => header * body))
+end
+
+function makedatatip(docs, word, mtable)
+  # @FIXME?: Separates code blocks from the other markdown texts in order to
+  #          render them as code snippet text by atom-ide-ui's datatip service.
+  #          Setting up functions to deconstruct each `Markdown.MD.content`
+  #          into an appropriate markdown string might be preferred.
+  texts = split(string(docs), codeblock_regex)
```

Comment:
Can the datatip package render HTML? If so it might be easiest to use [this](https://github.com/JunoLab/Atom.jl/blob/master/src/display/markdown.jl) and atom-julia-client's `views.render` together (or transform `Hiccup.Node`s into dicts, I guess).

Reply:
I tried to use them first, but the datatip packs only accept virtual doms made by react...

If we still stick on using our dom, we have to do [something trickly like this](https://github.com/TypeStrong/atom-typescript/blob/master/lib/main/atom-ide/datatipProvider.tsx).